### PR TITLE
fix: make auto retry value field mandatory

### PR DESCRIPTION
**Pull request by gitanjli525**
## Type of Change

- [x] Bugfix
- [ ] New feature
- [ ] Enhancement
- [ ] Refactoring
- [ ] Dependency updates
- [ ] Documentation
- [ ] CI/CD

## Description
- making the max auto retry value field mandatory in case of auto retry is enabled

image

## How did you test it?

1. Go to payment settings
2. enable auto retry toggle
3. try clicking update
4. update button shows error tooltip and is disabled

## Where to test it?

- [x] INTEG
- [x] SANDBOX
- [ ] PROD

## Checklist

- [x] I ran `npm run re:build`
- [x] I reviewed submitted code
- [ ] I added unit tests for my changes where possible

### CODE REVIEW

File: src/screens/Settings/MerchantAccountUtils.res [hunk 1]
```diff
@@ -419,7 +414,17 @@ let validateCustom = (key, errors, value, isLiveMode) => {
...
+  }
+}
 
+let validateEmptyValue = (key, errors) => {
+  switch key {
+  | MaxAutoRetries =>
+    errors->Dict.set(
+      key->validationFieldsMapper,
+      "Please enter a max auto retries value"->JSON.Encode.string,
+    )
```
Comment (Riddhiagrawal001) [inline src/screens/Settings/MerchantAccountUtils.res:427]: If the field is a numeric input then i guess you can add the condition in the same validate custom func

File: src/utils/LogicUtils.res [hunk 2]
```diff
@@ -233,6 +233,10 @@ let getOptionIntFromString = str => {
   str->Int.fromString
 }
 
+let getOptionStringFromInt = int => {
+  Some(int->Option.getOr(0)->Int.toString)
+}
+
```
Comment (Riddhiagrawal001) [inline src/utils/LogicUtils.res:239]: why it is always returning Some? If it is option string then why it is never returning none ?

File: src/screens/Developer/PaymentSettings/PaymentSettings.res [hunk 3]
```diff
@@ -376,9 +380,11 @@ let make = (~webhookOnly=false, ~showFormOnly=false, ~profileId="") => {
...
   let fieldsToValidate = () => {
     let defaultFieldsToValidate =
-      [WebhookUrl, ReturnUrl, MaxAutoRetries]->Array.filter(urlField =>
-        urlField === WebhookUrl || !webhookOnly
-      )
+      [
+        WebhookUrl,
+        ReturnUrl,
+        checkMaxAutoRetry ? MaxAutoRetries : UnknownValidateFields(""),
+      ]->Array.filter(urlField => urlField === WebhookUrl || !webhookOnly)
```
[the remark this answers is not in the document]
Comment (gitanjli525) [reply src/screens/Developer/PaymentSettings/PaymentSettings.res]: yeah sure

File: src/screens/Developer/APIKeys/DeveloperUtils.res [hunk 4]
```diff
@@ -192,7 +192,7 @@ let maxAutoRetries = FormRenderer.makeFieldInfo(
   ~name="max_auto_retries_enabled",
   ~placeholder="Enter number of max auto retries",
   ~customInput=InputFields.textInput(~autoComplete="off"),
```
[the remark this answers is not in the document]
Comment (gitanjli525) [reply src/screens/Developer/APIKeys/DeveloperUtils.res]: yes, corrected that

File: src/screens/Settings/MerchantAccountUtils.res (hunk 1 above)
Comment (gitanjli525) [reply src/screens/Settings/MerchantAccountUtils.res]: okay, updated

File: src/utils/LogicUtils.res (hunk 2 above)
Comment (gitanjli525) [reply src/utils/LogicUtils.res]: updated

**Review by sagarnaikjuspay — APPROVED** [review]

**Pull request merged**